frontend/ui/menusorter.lua

```diff
@@ -86,7 +86,10 @@ function MenuSorter:sort(item_table, order)
                 if v then
                     if v.id == separator_id then
                         new_index = new_index - 1
-                        menu_table[order_id][new_index].separator = true
+                        -- ignore separator if the menu starts with it
+                        if new_index > 0 then
```

Comment:
I'd say to use 1 instead of 0, just for the reader to sometimes be reminded than in lua, indexes start at 1 :)
(https://stackoverflow.com/questions/2785704/why-do-lua-arraystables-start-at-1-instead-of-0/2785854#2785854 to finger point petroleum engineers!)
But in your case, it seems 0 is a possible new_index value, so it may be unrelated to lua table indexes.